### ISSUE TITLE
Server with mod gives the "Error Adding Element" error 

### ISSUE THREAD

**Issue by Unusual-Danger** [nekomaster1000/Infernal-Expansion#261]
I'm trying to make a modded Minecraft server and get "Error Adding Element" error when the server finishes loading. Any way to fix this? 
[latest.log](https://github.com/infernalexp/Infernal-Expansion/files/6980228/latest.log)

**Comment by Edel-Lilies** [comment]
same,version 2.20

**Comment by LLjo** [comment]
Yep same. Caused severe lag on my server. Went back to 2.1 and it works good now

**Comment by faeldray** [comment]
Came by to say the same thing. It'll cause lag every 5 minutes or so and makes it annoying to play. I also noticed the same issue in single player although not quite as bad as on a server.

**Comment by caelwarner** [comment]
We're looking into it right now. We think it is related to #262. Is everyone playing with Quark installed?

**Comment by Unusual-Danger** [comment]
Yes I am playing with Quark installed

**Comment by faeldray** [comment]
I also have Quark installed but I did some further testing and was able to recreate the error with the following set of mods:

- Forge 36.1.61
- Abnormals Core 3.2.0
- Endergetic Expansion 3.0.0
- Infernal Expansion 2.2.0
- JEI 7.7.1.116
- The One Probe 3.1.4
- Xaero's Minimap 21.15.1
- Xaero's World Map 1.16.1

So I think the conflict lies with Endergetic Expansion.

**Comment by Edel-Lilies** [comment]
With Quark installed

**Comment by jodlodi** [comment]
I have the same issue, but I did some testing, when I was trying to find what mod was making the errors:
The Endergetic Expansion mod usually gives me only one "[Server thread/ERROR] [minecraft/WorldSettingsImport]: Error adding element: Base value out of range: 256 [-10-128]" error while loading up a world, but while I have Infernal Expansion installed, the error happens 4 times instead, each time the game is saved. The error does not happen at all without Endergetic Expansion on the other hand.

There is also an almost identical issue with another mod, valhelsia structures, where without Infernal Expansion installed an error only happens maybe once, but if both mods are installed, the error happens on each world save. (the error is not the same error as the one that happens with Endergetic Expansion, this one says Error adding element: Value 10 outside of range [0:7], instead)

**Comment by iMahco** [comment]
Adding onto the list: betterdefaultbiomes gives warnings whenever this mod (2.2.0) is introduced, the same lag spike as faeldray mentioned happens every 5mins, along with the following in console repeated 4 times back to back:
> [Server thread/INFO] [betterdefaultbiomes/]: Swamp trees added to: MINECRAFT:SWAMP_HILLS
> [Server thread/INFO] [betterdefaultbiomes/]: Swamp trees added to: MINECRAFT:SWAMP
> [Server thread/INFO] [betterdefaultbiomes/]: Palm trees added to: MINECRAFT:BEACH

Test environment was a clean install of forge 36.1.18 and 36.2.2
**Just** two mods installed: betterdefaultbiomes and infernalexpansion, no quark.
version 2.1.1 works just fine.

**Comment by wchen1990** [comment]
Pretty sure this is the issue: https://github.com/infernalexp/Infernal-Expansion/blob/0df577b872303e508a4dff270a47390584396d52/src/main/java/org/infernalstudios/infernalexp/mixin/common/MixinChunkGenerator.java#L29-L32

The ChunkGenerator is being replaced regardless of the config file settings and this is a massive change. Since this effects everything that uses the NoiseChunkGenerator.

I fixed this issue by just importing `org.infernalstudios.infernalexp.config.InfernalExpansionConfig.WorldGeneration` and adding an if statement to check if the `((boolean) WorldGeneration.REPLACE_NETHER_BIOME_PROVIDER.get())` is `true`.
```
    @Inject(at = @At("HEAD"), method = "<clinit>")
    private static void IE_clinit(CallbackInfo ci) {
        if ((boolean) WorldGeneration.REPLACE_NETHER_BIOME_PROVIDER.get()) {
            NoiseChunkGeneratorUtil.useCustomNetherBiomeProvider();
        }
    }
```

After making this change and testing the built jar on my local dev dedicated server I no longer get the errors.

**Comment by MrJorman** [comment]
my server really lagged..

**Comment by TheZennou** [comment]
Had to remove infernal expansion from my server, very disappointing, hope this is fixed soon.

**Comment by MrJorman** [comment]
i use 2.1.1 version.

**Comment by LordBoos** [comment]
Same problem on my server with 2.2.0. Reverting to 2.1.1 fixes the issue.

**Comment by SwanX1** [comment]
Fixed in 2b3fb682b9dc96f18492e7740a845fc99f34708e
<details>
<summary>Mentioned Users</summary>

 - @Unusual-Danger 
 - @Theoretical-Enday 
 - @zoyer2 
 - @faeldray 
 - @jodlodi 
 - @iMahco 
 - @wchen1990 
 - @MrJorman 
 - @The-Noble 
 - @LordBoos 
</details>

Please let me know if using this version of the mod doesn't fix your issue: [infernalstudios.org/etc/infernal-expansion-2.2.0+20210829.1.jar](https://infernalstudios.org/etc/infernal-expansion-2.2.0+20210829.1.jar)

**Comment by iMahco** [comment]
It did indeed @SwanX1, no more 5min-lag spikes!
